### PR TITLE
🤯 External wallet login

### PR DESCRIPTION
**Pull request by WRadoslaw**
- #4135

⚠️ Currently only flow works, the session will NOT persist between sessions

### CODE REVIEW

File: packages/atlas/src/components/_auth/ExternalSignInModal/ExternalSignInSteps/ExternalSignInModalEmailStep.tsx [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+  )
+  const { register, formState, handleSubmit } = useForm<{ email: string }>({
+    resolver: (data, ctx, options) => {
+      return zodResolver(
+        z.object({
+          email: z.string().email(),
+        })
+      )(data, ctx, options)
+    },
+  })
```
Comment (drillprop) [inline packages/atlas/src/components/_auth/ExternalSignInModal/ExternalSignInSteps/ExternalSignInModalEmailStep.tsx]: let's simplify this the same as in LoginModal

File: packages/atlas/src/components/_auth/ExternalSignInModal/ExternalSignInSteps/ExternalSignInModalMembershipsStep.tsx [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+      onClick: handleConfirm,
+    })
+  }, [handleConfirm, memberId, setPrimaryButtonProps])
+
+  return (
+    <ExternalSignInModalStepTemplate
+      title="Select membership"
+      subtitle="It looks like you have multiple memberships connected to this wallet. Select membership which you want to log in."
+      hasNavigatedBack={hasNavigatedBack}
+    >
```
Comment (drillprop) [inline packages/atlas/src/components/_auth/ExternalSignInModal/ExternalSignInSteps/ExternalSignInModalMembershipsStep.tsx:103]: Same suggestion as in sign in modal: https://github.com/Joystream/atlas/pull/4182#discussion_r1206750786

File: packages/atlas/src/components/_auth/ExternalSignInModal/ExternalSignInSteps/ExternalSignInModalMembershipsStep.tsx [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+          <StyledListItem
+            key={handle}
+            label={handle ?? 'Account'}
+            caption={shortenString(controllerAccount, 5)}
+            size={smMatch ? 'large' : 'medium'}
+            selected={memberId === id}
+            nodeStart={
+              <Avatar
+                size={40}
+                assetUrl={metadata?.avatar?.__typename === 'AvatarUri' ? metadata.avatar.avatarUri ?? '' : ''}
```
Comment (drillprop) [inline packages/atlas/src/components/_auth/ExternalSignInModal/ExternalSignInSteps/ExternalSignInModalMembershipsStep.tsx]: We have a helper function for this purpose. Use `getMemberAvatar`

File: packages/atlas/src/components/_auth/ExternalSignInModal/ExternalSignInSteps/ExternalSignInModalMembershipsStep.tsx (hunk 2 above)
Comment (WRadoslaw) [reply packages/atlas/src/components/_auth/ExternalSignInModal/ExternalSignInSteps/ExternalSignInModalMembershipsStep.tsx:103]: added to integration issue

**Review by drillprop — APPROVED** [review]

**Pull request merged**